enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,494 @@
+  -o, --output string                                        Result format for results. One of: [text, json-alpha1]. Note: output format types containing "alphaX" are subject to change and not covered by guarantees of stable APIs. (default "text")
+      --select-optional string                               Label selector to select optional validators to run. Run this command with '--list-optional' to list available optional validators
+
+Global Flags:
+      --plugins strings   plugin keys to be used for this subcommand execution
+      --verbose           Enable verbose logging
+```
+
+* *--help* works as is
+* *--image-builder* works as is
+* *--list-optional* would need to be updated to locate external validators.
+
+  TODO: How do we locate these validators? Do we use the XDG_CONFIG like Phase
+  2? Or allow users to pass in the location of the validators?
+
+* *--optional-values* would continue to be passed to the validators
+* *--output* indicates how we want to output the results.
+
+  TODO: what types do we want to support?
+
+* *--select-optional* takes in a label selector to specify which optional test
+  to run. We could allow a new label to indicate the location of the executable
+  to run as the validator. i.e. `validator-path=/path/to/validator/the-executable`
+
+TODO: need to define what the CLI for operator-sdk will look like. Are there
+any new flags? Do we use the environment variable?
+
+NOTE: what would the CLI look like?
+
+
+##### Default validators run by operator-sdk
+
+List of current validators:
+TODO: do we need these validators called out like this?
+
+* BundleValidator
+  * validates the bundle
+  * looks for duplicate keys in bundle
+  * verifies all owned keys match a CRD in the bundle
+  * verifies that all CRDs present in the bundle are in the CSV
+  * validates the service account
+
+* ClusterServiceVersionValidator
+  * operator-sdk checks to see if the CSV is nil on the bundle (that's the first
+    check)
+  * checks that the CSV name is a valid format
+    * DNS1123
+    * valid label
+  * verifies replaces name is also a valid format
+    * DNS1123
+    * valid label
+  * ensures that both `alm-examples` and `olm.examples` are not both present
+  * decodes the example yaml, to validate its format
+  * checks provided APIs
+  * validates the `InstallModes`
+    * verifies that conversion CRDs support `AllNamespaces`
+  * checks for missing mandatory fields
+  * validates the annotation names
+  * validates the version and kind
+
+* CustomResourceDefinitionValidator
+  * operator-sdk puts the v1beta1 and v1 CRDs together for validation
+  * validates v1beta1 CRDs
+  * validates v1 CRDs
+  * validates internal CRDs
+    * https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/validation/validation.go#L49-L78
+
+##### Optional validators
+
+* PackageManifestValidator
+
+* OperatorHubValidator
+
+* ObjectValidator
+
+* OperatorGroupValidator
+
+* CommunityOperatorValidator
```

Comment:
This one is vendor-like and we must remove it from operator-framework/API.
I'd like to propose we get started with an alpha version of the external validators to allow us:
- Remove these checks from the operator-framework/api
- Start to use and evaluate the solution for external validators

Reply:
I removed the list of existing validators. I didn't think they were adding any value to the EP.